cmd/skopeo/proxy.go

```diff
@@ -237,15 +248,45 @@ func isNotFoundImageError(err error) bool {
 		errors.Is(err, ocilayout.ImageNotFoundError{})
 }
 
-func (h *proxyHandler) openImageImpl(args []interface{}, allowNotFound bool) (replyBuf, error) {
+type openOptions struct {
+	optional bool
+}
+
+func (o *openOptions) parse(argsval interface{}) error {
+	args, ok := argsval.([]interface{})
+	if !ok {
+		return fmt.Errorf("expecting array for options, not %T", argsval)
+	}
+	for _, argv := range args {
+		arg, ok := argv.(string)
+		if !ok {
+			return fmt.Errorf("expecting string option, not %T", arg)
+		}
+		switch arg {
+		case "optional":
+			o.optional = true
+		default:
+			return fmt.Errorf("unknown option %s", arg)
+		}
+	}
+
+	return nil
+}
+
+func (h *proxyHandler) openImageImpl(args []interface{}, opts openOptions) (replyBuf, error) {
 	h.lock.Lock()
 	defer h.lock.Unlock()
 	var ret replyBuf
 
 	if h.sysctx == nil {
 		return ret, fmt.Errorf("client error: must invoke Initialize")
 	}
-	if len(args) != 1 {
+	switch len(args) {
+	case 1:
+		// This is is the default
+	case 2:
+		// The second argument, if present should have been parsed by the caller
```

Comment:
… but `OpenImageOptional` doesn’t do that…

I think enforcing `len(args)`, and turning `args` into better-typed data, should be collocated close to each other.

Probably in `OpenImage` (allowing `len(args) in {1,2}`) and `OpenImageOptional` (allowing `len(args) == 1`), calling already well-typed `openImageImpl(imageRef string, opts openOptions)`; or maybe have `openImageImpl` fully responsible for `args`, with the effect of `OpenImageOptional`  also allowing the new option argument. I’d prefer the former one but I don’t feel nearly as strongly about this choice as I do about having the the `len` check and the code that parses `argv` close.